source/MaterialXGenGlsl/GlslShaderGenerator.h

```diff
@@ -61,13 +61,22 @@ class MX_GENGLSL_API GlslShaderGenerator : public HwShaderGenerator
     virtual void emitVertexStage(const ShaderGraph& graph, GenContext& context, ShaderStage& stage) const;
     virtual void emitPixelStage(const ShaderGraph& graph, GenContext& context, ShaderStage& stage) const;
 
+    virtual void emitDirectives(GenContext& context, ShaderStage& stage) const;
+    virtual void emitConstants(GenContext& context, ShaderStage& stage) const;
+    virtual void emitUniforms(GenContext& context, ShaderStage& stage, HwResourceBindingContextPtr &resourceBindingCtx) const;
```

Comment:
Does `emitUniforms` need `HwResourceBindingContextPtr &resourceBindingCtx` or could it use `getResourceBindingContext`?

Reply:
Could be. I think Jerran added this so if the arg is not needed, we can remove it.

Reply:
Correction. This was introduced as part of the ESSL function refactoring. Removed the arg and made the emitLightData() also not have this arg. Both uset `getResourceBindingContext()` now.